src/overlays/actors/ovl_En_Railgibud/z_en_railgibud.c

```diff
@@ -65,122 +149,999 @@ static DamageTable D_80BA82CC = {
+    }
+}
+
+void func_80BA6284(EnRailgibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, 10);
+    this->actor.speedXZ = 0.4f;
+    this->actionFunc = func_80BA62D4;
+}
+
+void func_80BA62D4(EnRailgibud* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->unk_3E2, 0, 1, 0x64, 0);
+    Math_SmoothStepToS(&this->unk_3E8, 0, 1, 0x64, 0);
+    if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) < 5.0f) {
+        if (this->actor.speedXZ > 0.2f) {
+            this->actor.speedXZ -= 0.2f;
+        } else {
+            this->actor.speedXZ = 0.0f;
+        }
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 1, 0xC8, 0xA);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        if (this->actor.world.rot.y == this->actor.home.rot.y) {
+            func_80BA57A8(this);
+        }
+    } else {
+        Math_ScaledStepToS(&this->actor.shape.rot.y, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x1C2);
+        this->actor.world.rot = this->actor.shape.rot;
+    }
+    if (func_80BA6D10(this, globalCtx)) {
+        if ((gSaveContext.playerForm != PLAYER_FORM_GORON) && (gSaveContext.playerForm != PLAYER_FORM_DEKU) &&
+            Actor_IsActorFacingLink(&this->actor, 0x38E3)) {
+            func_80BA5AF0(this);
+        }
+    }
+}
+
+void func_80BA6440(EnRailgibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, 7);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_DAMAGE);
+    this->unk_3F2 = 0;
+    this->unk_3F4 = 0;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    this->actor.speedXZ = -2.0f;
+    this->actionFunc = func_80BA64AC;
+}
+
+void func_80BA64AC(EnRailgibud* this, GlobalContext* globalCtx) {
+    if (this->actor.speedXZ < 0.0f) {
+        this->actor.speedXZ += 0.15f;
+    }
+
+    if (func_801378B8(&this->skelAnime, this->skelAnime.animFrameCount)) {
+        this->unk_405 = -1;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        if ((this->unk_3F6 > 0) && (this->unk_404 == 0) && (this->unk_3F8 == 0)) {
+            this->actor.hintId = 0x2A;
+            SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06010B88, NULL, this->jointTable, this->morphTable, 26);
+            this->unk_3F8 = 1;
+        }
+        func_80BA6284(this);
+    }
+}
+
+void func_80BA6584(EnRailgibud* this) {
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->unk_3F2 = 10;
+    if (this->unk_3F6 != 0) {
+        func_800BCB70(&this->actor, 0, 0xC8, 0, 0x28);
+    } else {
+        func_800BCB70(&this->actor, 0, 0xC8, 0, 0x28);
+    }
+    this->actionFunc = func_80BA6604;
+}
+
+void func_80BA6604(EnRailgibud* this, GlobalContext* globalCtx) {
+    if (this->actor.colorFilterTimer == 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            func_80BA6664(this);
+        } else {
+            func_80BA6440(this);
+        }
+    }
+
+    if (this->unk_3F2 != 0) {
+        this->unk_3F2--;
+    }
```

Comment:
If this was my actor I would try turning this isn't a `DECR(this->unk_3F2)` but some seem against that

Reply:
Yeah I asked and apparently we shouldn't use it outside of ifs.